js/repeater.js

```diff
@@ -366,7 +366,7 @@
 
 			this.currentPage = (page !== undefined) ? page : NaN;
 
-			if ((this.currentPage + 1) >= pages) {
+			if (data.end || (this.currentPage + 1) >= pages) {
```

Comment:
I like this idea; however, the logic needs to be changed to `(data.end === true)`. That way the current way of handling it would still work, but this just simplifies things :)